recipes/msgpack/all/conanfile.py

```diff
@@ -49,33 +89,46 @@ def _configure_cmake(self):
         if self._cmake:
             return self._cmake
         self._cmake = CMake(self)
-        self._cmake.definitions["MSGPACK_BOOST"] = self.options.with_boost
+        self._cmake.definitions["MSGPACK_ENABLE_SHARED"] = self.options.shared
+        self._cmake.definitions["MSGPACK_ENABLE_STATIC"] = not self.options.shared
+        self._cmake.definitions["MSGPACK_ENABLE_CXX"] = self.options.cpp_api
+        self._cmake.definitions["MSGPACK_BOOST"] = self.options.get_safe("with_boost", False)
         self._cmake.definitions["MSGPACK_32BIT"] = self.settings.arch == "x86"
-        self._cmake.definitions["MSGPACK_BUILD_EXAMPLE"] = False
+        self._cmake.definitions["MSGPACK_BUILD_EXAMPLES"] = False
         self._cmake.definitions["MSGPACK_BUILD_TESTS"] = False
         self._cmake.configure(build_folder=self._build_subfolder)
         return self._cmake
 
     def build(self):
-        if not self.options.header_only:
+        if self.options.get_safe("with_boost") and \
+           (self.options["boost"].header_only or self.options["boost"].without_chrono or \
+            self.options["boost"].without_context or self.options["boost"].without_system or \
+            self.options["boost"].without_timer):
+            raise ConanInvalidConfiguration("msgpack with boost requires the following boost components: chrono, context, system and timer.")
+        if self.options.c_api:
             cmake = self._configure_cmake()
             cmake.build()
 
     def package(self):
         self.copy("LICENSE_1_0.txt", dst="licenses", src=self._source_subfolder)
-        if self.options.header_only:
-            self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
-            self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
-        else:
+        if self.options.c_api:
             cmake = self._configure_cmake()
             cmake.install()
             tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
             tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
-
-    def package_id(self):
-        if self.options.header_only:
-            self.info.header_only()
+        else:
+            self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+            self.copy("*.hpp", dst="include", src=os.path.join(self._source_subfolder, "include"))
 
     def package_info(self):
-        if not self.options.header_only:
-            self.cpp_info.libs = tools.collect_libs(self)
+        # TODO: CMake imported targets shouldn't be namespaced
```

Comment:
I need a link to conan's issue about that - it keeps repeating in many recipes

Reply:
you want a link to https://github.com/conan-io/conan/issues/7615 in this recipe?

Reply:
link added